pkg/stub/handler.go

```diff
@@ -242,6 +242,12 @@ func (h *Handler) ensureRouterForIngress(ci *ingressv1alpha1.ClusterIngress) err
 				} else if !errors.IsAlreadyExists(err) {
 					return fmt.Errorf("failed to create router service %s/%s: %v", service.Namespace, service.Name, err)
 				}
+				if ci.Spec.IngressDomain != nil {
+					err = h.ensureDNSForLoadBalancer(ci, service)
```

Comment:
we definitely need to think about how to rate limit how often this can be ensured (i.e. should we have a rate limiter that requeues us when the limit expires, should we have a periodic timer, etc).